devdoc/dynamic_sf_config_design.md

```diff
@@ -0,0 +1,531 @@
+# Dynamic Service Fabric Configuration - Detailed Design
+
+## Overview
+
+This document describes the design for adding dynamic configuration support to the Service Fabric configuration system in `sf-c-util`. The goal is to allow services to:
+
+1. Be notified when Service Fabric configuration packages change
+2. Re-read configuration values atomically using a reader-writer lock pattern
```

Comment:
>atomically [](http://example.com/codeflow?start=32&length=10)

sp: Automatically

Reply:
I think the bot meant atomically, as in we are either re reading all the config or none.